cmd/anubis/main.go

```diff
@@ -709,3 +716,16 @@ func serveMainJSWithBestEncoding(w http.ResponseWriter, r *http.Request) {
 	w.Header().Set("Content-Type", "text/javascript")
 	http.ServeFileFS(w, r, static, "static/js/main.mjs")
 }
+
+func newPrivateKey(secret string) (ed25519.PrivateKey, error) {
+	rr := rand.Reader
+	if secret != "" {
+		rr = hkdf.New(sha256.New, []byte(secret), nil, []byte("ed25519 signing key"))
+	}
```

Comment:
Note that secret should be a good random value (e.g. `head -c 32 /dev/urandom | base64`), HKDF is used for "key splitting", here to derive a signing key and maybe later to derive other keys from this (master) secret.

Alternative is to add "passphrase" flag and use password-based KDF (e.g. https://en.wikipedia.org/wiki/PBKDF2) to derive keys from it.